overlord/devicestate/devicemgr.go

```diff
@@ -1385,3 +1391,66 @@ func checkFDEFeatures(st *state.State, kernelInfo *snap.Info) error {
 	//       returns {"error":"hardware-unsupported"} we don't.
 	return nil
 }
+
+func (m *DeviceManager) runFDESetupHook(op string, params *boot.FdeSetupHookParams) ([]byte, error) {
+	hooksup := &hookstate.HookSetup{
+		Snap:     params.KernelInfo.InstanceName(),
+		Revision: params.KernelInfo.Revision,
```

Comment:
as we discussed before we cannot really run anything but the hook from the current installed kernel so probably we shouldn't use the KernelInfo from param that way but instead user snapstate.KernelInfo.

Reply:
indeed, silly me. I forgot to fix this, added now.